src/main/java/org/osc/controller/nsc/api/SampleSdnRedirectionApi.java

```diff
@@ -0,0 +1,329 @@
+            });
+        }
+    }
+
+    @Override
+    public Long getInspectionHookTag(NetworkElement inspectedPort, InspectionPortElement inspectionPort)
+            throws NetworkPortNotFoundException, Exception {
+        InspectionHookElement inspectionHook = getInspectionHook(inspectedPort, inspectionPort);
+        return inspectionHook == null ? null : inspectionHook.getTag();
+    }
+
+    @Override
+    public void setInspectionHookTag(NetworkElement inspectedPort, InspectionPortElement inspectionPort, Long tag)
+            throws Exception {
+
+        this.txControl.required(() -> {
+            InspectionHookEntity entity = this.utils.findInspHookByInspectedAndPort(inspectedPort, inspectionPort);
+            entity.setTag(tag);
+            return null;
+        });
+    }
+
+    @Override
+    public FailurePolicyType getInspectionHookFailurePolicy(NetworkElement inspectedPort,
+            InspectionPortElement inspectionPort) throws Exception {
+        InspectionHookElement inspectionHook = getInspectionHook(inspectedPort, inspectionPort);
+        return inspectionHook == null ? null : inspectionHook.getFailurePolicyType();
+    }
+
+    @Override
+    public void setInspectionHookFailurePolicy(NetworkElement inspectedPort, InspectionPortElement inspectionPort,
+            FailurePolicyType failurePolicyType) throws Exception {
+
+        this.txControl.required(() -> {
+            InspectionHookEntity entity = this.utils.findInspHookByInspectedAndPort(inspectedPort, inspectionPort);
+            entity.setFailurePolicyType(failurePolicyType);
+            return null;
+        });
+    }
+
+    @Override
+    public void removeAllInspectionHooks(NetworkElement inspectedPort) throws Exception {
+
+        this.txControl.required(() -> {
+            Query q = this.em.createQuery("FROM InspectionHookEntity WHERE inspectedPortId = :portId");
+            String portId = (inspectedPort != null ? inspectedPort.getElementId() : null);
+            q.setParameter("portId", portId);
+
+            @SuppressWarnings("unchecked")
+            List<InspectionHookEntity> results = q.getResultList();
+
+            for (InspectionHookEntity inspectionHookEntity : results) {
+                this.utils.removeSingleInspectionHook(inspectionHookEntity);
+            }
+
+            q = this.em.createQuery("FROM InspectionHookEntity WHERE inspectedPortId = :portId");
+            q.setParameter("portId", portId);
+            if (q.getMaxResults() > 0) {
+                LOG.error("Deleting inspection hooks for inspectedPortId failed! inspectedPortId: " + portId);
+            }
+            return null;
+        });
+
+    }
+
+    // Inspection port methods
+    @Override
+    public InspectionPortElement getInspectionPort(InspectionPortElement inspectionPort) throws Exception {
+
+        if (inspectionPort == null) {
+            return null;
+        }
+
+        try {
+            String portId = inspectionPort.getElementId();
```

Comment:
only try this if the  getElementId does not return null, if null just fallback to the port query in ln 212